client/components/purchase-detail/test/index.jsx

```diff
@@ -61,18 +70,24 @@ describe( 'PurchaseDetail', () => {
 			buttonText: 'test:button-text',
 		};
 
-		wrapper = shallow( <PurchaseDetail { ...buttonProps } /> );
-
-		const purchaseButton = wrapper.find( PurchaseButton );
-		expect( purchaseButton ).toHaveLength( 1 );
-		expect( purchaseButton.prop( 'disabled' ) ).toBe( false );
-		expect( purchaseButton.prop( 'href' ) ).toEqual( 'https://wordpress.com/test/url' );
-		expect( purchaseButton.prop( 'onClick' ) ).toEqual( noop );
-		expect( purchaseButton.prop( 'target' ) ).toEqual( 'test:target' );
-		expect( purchaseButton.prop( 'rel' ) ).toEqual( 'test:rel' );
-		expect( purchaseButton.prop( 'text' ) ).toEqual( buttonProps.buttonText );
+		const { rerender } = render( <PurchaseDetail { ...buttonProps } /> );
+
+		const purchaseButton = screen.queryByTestId( 'purchase-button' );
+		expect( purchaseButton ).toBeVisible();
+
+		expect( PurchaseButton ).toHaveBeenCalledWith(
```

Comment:
Same as above, but for `PurchaseButton`, which we're also mocking already anyway.